crates/common_utils/src/id_type/ephemeral_key.rs

```diff
@@ -0,0 +1,30 @@
+crate::id_type!(
+    EphemeralKeyId,
+    "A type for key_id that can be used for Ephemeral key IDs"
+);
+crate::impl_id_type_methods!(EphemeralKeyId, "key_id");
+
+// This is to display the `EphemeralKeyId` as EphemeralKeyId(abcd)
+crate::impl_debug_id_type!(EphemeralKeyId);
+crate::impl_try_from_cow_str_id_type!(EphemeralKeyId, "key_id");
+
+crate::impl_serializable_secret_id_type!(EphemeralKeyId);
+crate::impl_queryable_id_type!(EphemeralKeyId);
+crate::impl_to_sql_from_sql_id_type!(EphemeralKeyId);
+
+impl EphemeralKeyId {
+    /// Generate Ephemeral Key Id from prefix
+    pub fn generate_key_id(prefix: &'static str) -> Self {
+        Self(crate::generate_ref_id_with_default_length(prefix))
+    }
```

Comment:
If we know the prefix at compile time, we can directly use it with the `impl_generate_id_id_type` macro.